clienr-rec/src/components/FilterButtons.jsx

```diff
@@ -0,0 +1,33 @@
+import React from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { setFilter } from "../features/tasks/taskSlice";
+
+const FilterButtons = () => {
+  const dispatch = useDispatch();
+  const filter = useSelector((state) => state.tasks.filter);
+
+  return (
+    <div className="flex gap-2 mb-4 mt-6">
+      <button
+        className={`p-2 rounded ${filter === "all" ? "bg-gray-500 text-white" : "bg-gray-300"}`}
+        onClick={() => dispatch(setFilter("all"))}
+      >
+        All
+      </button>
+      <button
+        className={`p-2 rounded ${filter === "completed" ? "bg-green-500 text-white" : "bg-green-300"}`}
+        onClick={() => dispatch(setFilter("completed"))}

```

Comment:
Please define all, completed and pending as json based constants and utilize the json params to use in the code. 
```
const status = {
all: 'all',
completed: 'completed',
pending: 'pending',
}
```

```
 onClick={() => dispatch(setFilter(status.completed))
```